lib/utils.js

```diff
@@ -27,7 +27,7 @@ export function isMajorRange(range) {
 }
 
 export function isMaintenanceRange(range) {
-  return /^\d+\.(?:\d+|x)(?:\.x)?$/i.test(range);
+  return /^\d+(?:\.(\d+|x))?(?:\.x)$/i.test(range);
```

Comment:
to help explain this change as we look back at it in the future, the key detail about this change is that it makes the second segment the optional one rather than the third segment. that ensures that the resulting version-ish branch name is a range when considered to be a match for a maintenance branch

Reply:
I'm going to add this explaination to the commit message

Reply:
if you are updating the commit message, lets also make the change summary a little more targeted at the reason for this. something like `fix(branch-naming): prevent non-range versions from being identified as maintenance branches` seems more descriptive to me